packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -247,6 +254,30 @@ abstract class HttpApiBase extends Resource implements IHttpApi { // note that t
 
     return vpcLink;
   }
+
+  /**
+   * @internal
+   */
+  public _addIntegration(config: HttpRouteIntegrationConfig): HttpIntegration {
+    const configKey = JSON.stringify(config);
+    if (configKey in this.httpIntegrations) {
+      return this.httpIntegrations[configKey];
+    }
+
+    const count = Object.keys(this.httpIntegrations).length + 1;
+    const integration = new HttpIntegration(this, `HttpIntegration-${count}`, {
```

Comment:
Use a hash of `configKey` instead of count.

Something like this - https://github.com/aws/aws-cdk/blob/2a032eaec22c01489ff567f51bba0d85580f944c/packages/%40aws-cdk/aws-lambda/lib/function-hash.ts#L15-L17